website/blog/2023-03-08-mongodb-vs-postgresql.md

```diff
@@ -0,0 +1,181 @@
+
+### Data Model
+
+Regarding data models, MongoDB and PostgreSQL follow different approaches, with MongoDB using a document-based data model, while PostgreSQL uses a relational data model.
+
+Each piece of data in MongoDB is stored as a JSON-like document containing dynamic schemas – each document can be structured differently, and new fields can be added at any time using key-value pairs.
+This makes MongoDB perfect for scenarios where the data constantly changes and there's no defined data structure, such as social media platforms where users can upload their own content.
+
+Contrast this with PostgreSQL, which requires a defined database schema and is used to store data in traditional table format with predefined columns and data types.
+PostgreSQL is more suitable for applications that need to maintain relationships among different data types, such as e-commerce sites or financial applications.
+
+### Query Language
+
+MongoDB and PostgreSQL use different query languages, which are pretty different in syntax and functionality.
+
+To illustrate this difference, PostgreSQL uses Structured Query Language (SQL) to query relational databases composed of multiple tables with a defined relationship between them.
+SQL is a widely known language that cuts across many great tools and platforms, including Oracle and MySQL.
+SQL databases enable you to write complex queries and run different operations on relational data.
+
+Below is an example of a typical SQL query that selects all columns and prints out all records from the `user` table.
+
+```js
+SELECT * FROM users;
+```
+
+Unlike PostgreSQL, MongoDB does not support SQL queries natively.
+Instead, it has its own query language – MongoDB Query Language (MQL).
+MQL is built to interact specifically with MongoDB databases and match similar features and flexibility as in SQL databases.
+With MQL, you can query any field, embedded documents, or nested arrays in the MongoDB database.
+
+```js
+db.users.find({
+  hobbies: { $all: ["reading", "cooking"] }
+})
+```
+
+### Performance and Scalability
+
+Comparing the performance of MongoDB and PostgreSQL is a complex task due to their distinctive approaches to data storage and retrieval.
+
+PostgreSQL's rigid schema and strong typing may result in slower inserts and updates due to server-side type checking and schema validation.
+However, PostgreSQL's support for relations and JOINs allows users to create complex, structured data models that can return organized data from multiple tables with a single JOIN query.
+
+PostgreSQL uses a vertical scaling strategy to Manage vast amounts of data and increase write scalability by adding hardware resources such as disks, CPUs, and memory to existing database nodes.
+
+Although PostgreSQL may not match the raw insertion speed of MongoDB, its exceptional ACID compliance ensures safe and reliable transaction processing, preventing partial writes by executing entire transactions or none at all.
```

Comment:
I would put ACID section above _Performance and Scalability_ section because we mention ACID in this sentence for the first time.